src/http.zig

```diff
@@ -2368,12 +2417,37 @@ fn printResponse(response: picohttp.Response) void {
     Output.flush();
 }
 
+pub fn onPreconnect(this: *HTTPClient, comptime is_ssl: bool, socket: NewHTTPContext(is_ssl).HTTPSocket) void {
+    log("onPreconnect({})", .{this.url});
+    _ = socket_async_http_abort_tracker.swapRemove(this.async_http_id);
+    const ctx = if (comptime is_ssl) &http_thread.https_context else &http_thread.http_context;
+
+    ctx.releaseSocket(
```

Comment:
Can we add a test when we preconnect, but we close the connection on the server part before we do a fetch request?

Reply:
yeah, though the rest of the existing keep-alive logic is the same